cookbook/deployment/cluster/auth_setup.py

```diff
@@ -1,8 +1,498 @@
+
+  kubectl edit secret -n flyte flyte-propeller-auth
+
+Add a new key under `stringData`:
+
+.. code-block:: yaml
+
+  stringData:
+    client_secret: <client_secret> from the previous step
+  data:
+    ...
+
+Save and close your editor.
+
+2. Edit FlytePropeller config to add `client_id` and configure auth as follows:
+
+.. prompt:: bash
+
+  kubectl get deploy -n flyte flytepropeller -o yaml | grep "name: flyte-propeller-config"
+
+This will output the name of the config map where the `client_id` needs to go.
+
+.. prompt:: bash
+
+  kubectl edit configmap -n flyte <the name of the config map from previous command>
+
+Follow the inline comments to make the necessary changes:
+
+.. code-block:: yaml
+
+    admin:
+        # 1. Replace with the client_id provided by the OAuth2 Authorization Server above.
+        clientId: flytepropeller
+
+Close the editor
+
+3. Restart `flytepropeller` for the changes to take effect:
+
+.. prompt:: bash
+
+  kubectl rollout restart deployment/flytepropeller -n flyte
+
+***************************
+Continuous Integration - CI
+***************************
+
+If your organization does any automated registration, then you'll need to authenticate with the `basic authentication <https://tools.ietf.org/html/rfc2617>`_ flow (username and password effectively). After retrieving an access token from the IDP, you can send it along to Flyte Admin as usual.
+
+Flytekit configuration variables are automatically designed to look up values from relevant environment variables. However, to aid with continuous integration use-cases, Flytekit configuration can also reference other environment variables. 
+
+For instance, if your CI system is not capable of setting custom environment variables like ``FLYTE_CREDENTIALS_CLIENT_SECRET`` but does set the necessary settings under a different variable, you may use ``export FLYTE_CREDENTIALS_CLIENT_SECRET_FROM_ENV_VAR=OTHER_ENV_VARIABLE`` to redirect the lookup. A ``FLYTE_CREDENTIALS_CLIENT_SECRET_FROM_FILE`` redirect is available as well, where the value should be the full path to the file containing the value for the configuration setting, in this case, the client secret. We found this redirect behavior necessary when setting up registration within our own CI pipelines.
+
+The following is a listing of the Flytekit configuration values we set in CI, along with a brief explanation.
+
+* ``FLYTE_CREDENTIALS_CLIENT_ID`` and ``FLYTE_CREDENTIALS_CLIENT_SECRET``
+  When using basic authentication, this is the username and password.
+* ``export FLYTE_CREDENTIALS_AUTH_MODE=basic``
+  This tells the SDK to use basic authentication. If not set, Flytekit will assume you want to use the standard OAuth based three-legged flow.
+* ``export FLYTE_CREDENTIALS_AUTHORIZATION_METADATA_KEY=text``
+  At Lyft, the value is set to conform to this `header config <https://github.com/flyteorg/flyteadmin/blob/eaca2fb0e6018a2e261e9e2da8998906477cadb5/pkg/auth/config/config.go#L53>`_ on the Admin side.
```

Comment:
Is this statement about Lyft still valid?

Reply:
yeah